katportalclient/client.py

```diff
@@ -1254,73 +1193,40 @@ def sensor_history(self, sensor_name, start_time_sec, end_time_sec,
+        params['additional_fields'] = additional_fields
         url = url_concat(
-            self.sitemap['historic_sensor_values'] + '/samples', params)
+            self.sitemap['historic_sensor_values'] + '/query', params)
         self._logger.debug("Sensor history request: %s", url)
-        response = yield self._http_client.fetch(url)
-        data = json.loads(response.body)
-        if isinstance(data, dict) and data['result'] == 'success':
-            download_start_sec = time.time()
-            # Query accepted by portal - data will be returned via websocket, but
-            # we need to wait until it has arrived.  For synchronisation, we wait
-            # for a 'done_event'. This event is updated in
-            # _process_redis_message().
-            try:
-                timeout_delta = timedelta(seconds=timeout_sec)
-                yield state['done_event'].wait(timeout=timeout_delta)
-
-                self._logger.debug('Done in %d seconds, fetched %s samples.' % (
-                    time.time() - download_start_sec,
-                    len(state['samples'])))
-            except tornado.gen.TimeoutError:
-                raise SensorHistoryRequestError(
-                    "Sensor history request timed out")
-
-        else:
+        response = yield self._http_client.fetch(url, request_timeout=timeout_sec, 
+                                                 connect_timeout=timeout_sec)
+        data_json = json.loads(response.body)
+        if 'data' not in data_json:
             raise SensorHistoryRequestError("Error requesting sensor history: {}"
                                             .format(response.body))
-
-        def sort_by_timestamp(sample):
-            return sample.timestamp
-        # return a sorted copy, as data may have arrived out of order
-        result = sorted(state['samples'], key=sort_by_timestamp)
-
-        if len(result) >= MAX_SAMPLES_PER_HISTORY_QUERY:
-            self._logger.warn(
-                'Maximum sample limit (%d) hit - there may be more data available.',
-                MAX_SAMPLES_PER_HISTORY_QUERY)
-
-        # Free the state variables that were only required for the duration of
-        # the download.  Do not disconnect - there may be websocket activity
-        # initiated by another call.
-        yield self.unsubscribe(namespace, ['*'])
-        del self._sensor_history_states[namespace]
-
+        data = []
+        for item in data_json['data']:
+            if 'value_time' in item:
+                sensor = SensorSampleValueTs(item['sample_time'],
+                                             item['value_time'],
+                                             item['value'],
+                                             item['status'])
+            else:
+                sensor = SensorSample(item['sample_time'],
+                                      item['value'],
+                                      item['status'])
+            data.append(sensor)
```

Comment:
The items being appended are sensor samples, not sensors, so `sample` seems like a clearer name than `sensor`.